cpp/oneapi/dal/io/csv/detail/read_graph_kernel_impl.hpp

```diff
@@ -33,21 +33,49 @@ namespace oneapi::dal::preview::csv::detail {
 template <typename EdgeList>
 inline void read_edge_list(const std::string &name, EdgeList &elist);
 
-template <>
-inline void read_edge_list(const std::string &name, edge_list<std::int32_t> &elist) {
+template <typename Vertex>
+inline void read_edge_list(const std::string &name, edge_list<Vertex> &elist) {
     std::ifstream file(name);
     if (!file.is_open()) {
         throw invalid_argument(dal::detail::error_messages::file_not_found());
     }
-
     elist.reserve(1024);
-    char source_vertex[32], destination_vertex[32];
-    while (file >> source_vertex >> destination_vertex) {
-        auto edge = std::make_pair(daal_string_to_int(&source_vertex[0], 0),
-                                   daal_string_to_int(&destination_vertex[0], 0));
+
+    std::string line;
+    char *source_endptr;
+    char *dest_endptr;
+    const char *edgeline;
+    std::int32_t source_vertex;
```

Comment:
Now Vertex template was introduced [line 36]. Is it possible to change std::int32_t to Vertex (or what is a reasoning for not changing it)?

Reply:
Thanks! I missed it